### PR TITLE
Remove test/fuzzer/sqlsmith/timestamptz_double_cast.test

### PR DESCRIPTION
**Pull request by carlopi**
Test was hanging, causing OSX CI to fail. Removing this fixed the problem, verified on my fork.

Also removed the dependencies between jobs in OSX workflow, since this speeds up an already lengthy workflow.

### CODE REVIEW

File: test/fuzzer/sqlsmith/timestamptz_double_cast.test [hunk 1]
```diff
@@ -1,25 +0,0 @@
-# name: test/fuzzer/sqlsmith/timestamptz_double_cast.test
```
[the remark this answers is not in the document]
Comment (carlopi) [reply test/fuzzer/sqlsmith/timestamptz_double_cast.test:1]: You are probably right, I was also on the fence on what was the right approach here.

Issue as been assessed, @hawkfish took a deep dive, and he found some weird thing that needed to be fixed.

Given the result of his investigation, it's obvious that `mode skip` is better, will change the PR accordingly.

**Comment by carlopi** [comment]
Fixed the proper way in https://github.com/duckdb/duckdb/pull/8679

File: .github/workflows/OSX.yml [hunk 2]
```diff
@@ -75,7 +75,6 @@ jobs:
     # Builds binaries for osx_arm64 and osx_amd64
     name: OSX Release
     runs-on: macos-latest
-    needs: xcode-debug
```
Comment (hannes) [inline .github/workflows/OSX.yml:78]: is this on purpose?

File: test/fuzzer/sqlsmith/timestamptz_double_cast.test (hunk 1 above)
[the remark this answers is not in the document]
Comment (hannes) [reply test/fuzzer/sqlsmith/timestamptz_double_cast.test:1]: +1

File: .github/workflows/OSX.yml (hunk 2 above)
Comment (carlopi) [reply .github/workflows/OSX.yml:78]: This was on purpose, but this PR has no reason to be here now that https://github.com/duckdb/duckdb/pull/8679 is done.

**Pull request closed without merging**